aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1161,10 +1203,11 @@ AT_ERROR("cholesky: MAGMA library not found in "
 
     MAGMAQueue magma_queue(self.get_device());
 
-    constexpr int64_t batch_limit = 262140;
+    int64_t batch_limit = self.is_complex() ? 65535 : 262140;
```

Comment:
why do we have different batch limit for complex and non-complex dtypes? can you link me to where this is documented?

Reply:
I don't know whether it's documented somewhere, I determined this value via experiments.

Reply:
CUDA limits kernel launches to y and z grid dimension to 65535. Maybe for non-complex dtypes batching is implemented differently.

Reply:
synced with @ngimel offline. We should check the magma manual, and better document this difference in the `batch_limit` since the original comments are uninformative.

Reply:
It's not documented in magma.
CUDA limits kernel launch configurations of y and z grid dimensions to 65535.
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications
Maximum x-dimension of a grid of thread blocks | 2^31-1
-- | --
Maximum y- or z-dimension of a grid of thread blocks | 65535

I haven't checked the source code for how batching is done for non-complex dtypes, but apparently, complex variants use z-dimension of a grid of thread blocks for batching.

Reply:
@ngimel what should we do here?

Reply:
If the tests are passing we can leave it as is, though I still don't understand how bigger batch size works for real dtypes.

Reply:
I spent a little time looking at the complex path and didn't figure it out, but I did see this:

```
    if ( n > 2048 ) {   
        #ifndef MAGMA_NOWARNING
        printf("=========================================================================================\n"
               "   WARNING batched routines are designed for small sizes. It might be better to use the\n"
               "   Native/Hybrid classical routines if you want good performance.\n"
               "=========================================================================================\n");
        #endif
    }           
            
```

in `magma_cpotrf_lg_batched`

Reply:
Yeah we should use cusolver for those, if we don't already.

Reply:
cc @heitorschueroff, @xwang233 Can you guys please create a tracking issue which linalg functions under which conditions use magma or cusolver or cublas, and which functions still need to be weaned off magma and switched to cusolver?

Reply:
Thanks, I'll create a tracking issue.

Reply:
@xwang233's issue is here: https://github.com/pytorch/pytorch/issues/47953